src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java

```diff
@@ -997,6 +1000,9 @@ private void readFilesInfo(final ByteBuffer header, final Archive archive) throw
                         throw new IOException("Unimplemented");
                     }
                     for (int i = 0; i < files.length; i++) {
+                        if (files[i] == null) {
+                            files[i] = new SevenZArchiveEntry();
+                        }
```

Comment:
Agree with @garydgregory about this duplicate code. We can have a simple refactor here.

Reply:
Code is refactored as part of the latest commit